tools/check_reordering.py

```diff
@@ -0,0 +1,167 @@
+            continue
+
+        if symbolInfo.diff == 0:
+            symbolGood = colorama.Fore.GREEN + "GOOD" + colorama.Fore.RESET
+            if not buildFile in comparisonInfo.badFiles and not expectedFile in comparisonInfo.badFiles:
+                if not buildFile in comparisonInfo.badFiles and not expectedFile in comparisonInfo.badFiles:
+                    if not printAll:
+                        continue
+
+        if buildFile != expectedFile:
+            symbolGood += colorama.Fore.CYAN + " MOVED" + colorama.Fore.RESET
+        print(f"{symbolInfo.symbol.name},{symbolInfo.buildAddress:X},{buildFileName},{symbolInfo.expectedAddress:X},{expectedFileName},{symbolInfo.diff:X},{symbolGood}")
+
+
+def printFileComparison(comparisonInfo: mapfile_parser.MapsComparisonInfo, fun_allowed: bool = True):
+    eprint("")
+
+    if len(comparisonInfo.badFiles) != 0:
+        print(colorama.Fore.RED + "  BAD" + colorama.Style.RESET_ALL)
+
+        for file in comparisonInfo.badFiles:
+            eprint(f"Symbol reordering in {mapPathToSource(file.filepath)}")
+        eprint("")
+
+        if fun_allowed:
+            eprint(colorama.Fore.LIGHTWHITE_EX +
+            "  Symbols are REORDERED!!\n"
+            "  Oh! MY GOD!!" 
+            + colorama.Style.RESET_ALL)
+            eprint("")
+
+    if len(comparisonInfo.missingFiles) != 0:
+        print(colorama.Fore.YELLOW + "  MISSING" + colorama.Style.RESET_ALL)
+
+        for file in comparisonInfo.missingFiles:
+            eprint(f"Symbols missing from {mapPathToSource(file.filepath)}")
+        eprint("")
+
+        if fun_allowed:
+            eprint(colorama.Fore.LIGHTWHITE_EX + "  Error, should (not) be in here " + colorama.Style.RESET_ALL)
+            eprint("")
+
+        eprint("Some files appear to be missing symbols. Have they been renamed or declared as static? You may need to remake 'expected'")
+
+
+def main():
+    description = "Check that globally visible symbols has not been reordered."
+    epilog = """\
+    N.B. Since this script reads the map files, it can only see globally visible symbols; in-function static symbols must be examined with other tools.
+    """
+
+    parser = argparse.ArgumentParser(description=description, epilog=epilog, formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("-a", "--print-all", help="Print all symbols of the section, not just non-matching.", action="store_true")
+    parser.add_argument("-n", "--no-fun-allowed", help="Remove amusing messages.", action="store_true")
+    parser.add_argument("-r", "--no-reverse-check", help="Disable looking for symbols on the expected map that are missing on the built map file.", action="store_true")
+    parser.add_argument("-s", "--section", help="Specify which section should be checked for reordered symbols. Use * to check all sections. Defaults to .bss", default=".bss")
```

Comment:
I wonder if the script should except `all` to check all sections, rather than "*" just to avoid `*` being the wildcard in bash as well.

Reply:
it is a standard wildcard symbol though, including in linker scripts and binutils section manipulation
I think it's fine to expect the user knows to quote it like `--section='*'`
I'm not even sure `blabla=*` would cause the shell to expand `*` to anything tbh

Reply:
Shell does expand the `*`
```
$ tools/check_reordering.py --section *
usage: check_reordering.py [-h] [-a] [-n] [-r] [-s SECTION]
check_reordering.py: error: unrecognized arguments: assets baserom baserom.mm.us.rev1.z64 baseroms baserom_uncompressed.z64 build ...
```
Doing `--section=*` and `-s=*` do not expand to everything on the root folder

Given that this is kinda bad I think I'll go with `all` instead